airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -654,8 +673,8 @@ class SageMakerTrainingOperator(SageMakerBaseOperator):
         the operation does not timeout.
     :param check_if_job_exists: If set to true, then the operator will check whether a training job
         already exists for the name in the config.
-    :param action_if_job_exists: Behaviour if the job name already exists. Possible options are "increment"
-        (default) and "fail".
+    :param action_if_job_exists: Behaviour if the job name already exists. Possible options are "random"
+        (default), "increment" (deprecated) and "fail".
```

Comment:
I may have missed it. Where did we raise deprecation warning for this value?

Reply:
I didn't because the behavior is the same as "random" in my proposed code, so there is really no specific action required from users, I'd just prefer new code to use the "right" value.
But if we want to maintain the old behavior, then I'd add a warning

Reply:
My concern is specifically about pointing the (deprecated)
usually if we deprecate something, while immediate action isn't required because we are backward compatible an action is required to be done for the next major release (when we are making breaking change). so if we say something is deprecated we must also say what to do, otherwise this is just inviting many questions from users who are not sure what is expected from them.

Reply:
so, I don't think that's the direction this PR is taking, but if I were to keep the code as-is, changing this to `"increment" (deprecated, use "random" instead)` would be good for you ?

Reply:
No, my comment is that we consider value deprecated yet the code does not raise deprecation warning.
If something is deprecated it must have deprecation warning that lead users to set the proper values/parameters.

Reply:
added deprecation warnings